rkt/fetch.go

```diff
@@ -47,9 +59,106 @@ func init() {
 	cmdFetch.Flags().Var((*appAsc)(&rktApps), "signature", "local signature file to use in validating the preceding image")
 	cmdFetch.Flags().BoolVar(&flagStoreOnly, "store-only", false, "use only available images in the store (do not discover or download from remote URLs)")
 	cmdFetch.Flags().BoolVar(&flagNoStore, "no-store", false, "fetch images ignoring the local store")
+	cmdFetch.Flags().IntVar(&flagP2pDuration, "p2p-duration", 10, "p2p continue service duration (minutes) after download finished")
+}
+
+func p2pFetch(args []string) int {
+	if len(args) < 1 {
+		fmt.Printf("rkt fetch must provide a tottent file.")
+		return 1
+	}
+
+	duration := strconv.Itoa(flagP2pDuration)
+	//start p2p client
+	var cmd *exec.Cmd
+	cmd = exec.Command("./torrent", args[0], duration)
+	if err := cmd.Start(); err != nil {
+		fmt.Printf("start p2p process err:%s\n", err)
+		return 1
+	}
+
+	pid := cmd.Process.Pid
+
+	//handle Ctrl+C signal
+	go func() {
+		signalChan := make(chan os.Signal, 1)
+		signal.Notify(signalChan, os.Interrupt, os.Kill)
+
+		<-signalChan
+		if err := syscall.Kill(pid, 9); err != nil {
+			fmt.Printf("kill p2p client process err:%s\n", err)
+		}
+	}()
+
+	//wait for p2p client start
+	time.Sleep(time.Second * time.Duration(7))
+
+	var (
+		client  *rpc.Client
+		err     error
+		s       *store.Store
+		aciFile *os.File
+		key     string
+	)
+	//connet to p2p client process and get download rate
+	if client, err = rpc.DialHTTP("tcp", "127.0.0.1:1234"); err != nil {
+		fmt.Printf("connet to p2p client err:%s\n", err)
+		return 1
+	}
+	reply := make([]string, 1)
+	if err = client.Call("Download.GetDownloadTotalSize", struct{}{}, &reply); err != nil {
+		fmt.Printf("get download total size err:%s\n", err)
+		return 1
+	}
+	totalSize := reply[0]
+
+	if err = client.Call("Download.GetDownloadFile", struct{}{}, &reply); err != nil {
+		fmt.Printf("get download file err:%s\n", err)
+		return 1
+	}
+	aciImage := reply[0]
+
+	//get rate for download
+	for {
+		if err = client.Call("Download.GetDownloadRate", struct{}{}, &reply); err != nil {
+			fmt.Printf("get download rate err:%s\n", err)
+			return 1
+		}
+		fmt.Printf("\rtotal size:%sKB, downloaded:%s", totalSize, reply)
+		if reply[0] == "100.00%" {
+			break
+		}
+		time.Sleep(time.Second * time.Duration(5))
+	}
+
+	//save aci to rkt store
+	if s, err = store.NewStore(globalFlags.Dir); err != nil {
+		fmt.Printf("open rkt store err:%s\n", err)
+		return 1
+	}
+	if aciFile, err = os.Open(aciImage); err != nil {
+		fmt.Printf("opening ACI file %s err:%s\n", aciImage, err)
+		return 1
+	}
+	if key, err = s.WriteACI(aciFile, true); err != nil {
+		fmt.Printf("write ACI file err:%s\n", err)
+		return 1
+	}
+	fmt.Println(key)
```

Comment:
Is this a debug print? IMO this should be removed or dependent on some Args flag.